t/lib/t/MusicBrainz/Server/Controller/Label/AddAlias.pm

```diff
@@ -5,69 +5,122 @@ use MusicBrainz::Server::Test qw( capture_edits html_ok );
-});
-
-$mech->get_ok('/edit/' . $edit->id, 'Fetch edit page');
-html_ok($mech->content);
-
-$mech->content_contains('Warp Records', '..contains label name');
-$mech->content_contains('/label/46f0f4cd-8aab-4b33-b698-f459faf64190', '..contains label link');
-$mech->content_contains('An alias', '..contains alias name');
-$mech->content_contains('An alias sort name', '..contains alias sort name');
-
-# A sortname isn't required (MBS-6896)
-($edit) = capture_edits {
-    $mech->get_ok('/label/46f0f4cd-8aab-4b33-b698-f459faf64190/add-alias');
-    $mech->submit_form(
-        with_fields => {
-            'edit-alias.name' => 'Another alias',
-        });
-} $c;
-
-isa_ok($edit, 'MusicBrainz::Server::Edit::Label::AddAlias');
-is($edit->data->{sort_name}, 'Another alias', 'sort_name defaults to name');
+=head2 Test description
 
+This test checks whether alias adding for labels works, including whether
+the sort name defaults to the name when not explicitly entered.
+
+=cut
+
+test 'Test adding alias with sort name' => sub {
+    my $test = shift;
+    my $mech = $test->mech;
+
+    prepare_test($test);
+
+    $mech->get_ok(
+        '/label/46f0f4cd-8aab-4b33-b698-f459faf64190/add-alias',
+        'Fetched the add alias page',
+    );
+
+    my @edits = capture_edits {
+        $mech->submit_form_ok({
+            with_fields => {
+                'edit-alias.name' => 'An alias',
+                'edit-alias.sort_name' => 'An alias sort name',
+            },
+        },
+        'The form returned a 2xx response code')
+    } $test->c;
+
+    is(@edits, 1, 'The edit was entered');
+
+    my $edit = shift(@edits);
+    isa_ok($edit, 'MusicBrainz::Server::Edit::Label::AddAlias');
+
+    is_deeply(
+        $edit->data,
+        {
+            entity => {
+                id => 2,
+                name => 'Warp Records'
+            },
+            name => 'An alias',
+            sort_name => 'An alias sort name',
+            locale => undef,
+            primary_for_locale => 0,
+            begin_date => {
+                year => undef,
+                month => undef,
+                day => undef
+            },
+            end_date => {
+                year => undef,
+                month => undef,
+                day => undef
+            },
+            type_id => undef,
+            ended => 0,
+        },
+        'The edit contains the right data',
+    );
+
+    $mech->get_ok('/edit/' . $edit->id, 'Fetched the edit page');
+    html_ok($mech->content);
+
+    $mech->content_contains('Warp Records', 'Edit page contains label name');
+    $mech->content_contains(
+        '/label/46f0f4cd-8aab-4b33-b698-f459faf64190',
+        'Edit page contains artist link',
+    );
+    $mech->content_contains('An alias', 'Edit page contains alias name');
+    $mech->content_contains(
+        'An alias sort name',
+        'Edit page contains the selected alias sort name',
+    );
 };
 
+test 'MBS-6896: Test adding alias without sort name' => sub {
```

Comment:
For test descriptions with ticket numbers, I think they should describe what the issue or desired behavior is.  For example, let's say we wanted the page to throw an error if you don't give a sort name.  This description is a bit too vague because, since it doesn't say what should happen, it could apply to either situation.

```suggestion
test 'MBS-6896: Adding alias without sort name defaults it to the name' => sub {
```